libs/utils/utils.js

```diff
@@ -261,6 +261,64 @@ export const [setConfig, updateConfig, getConfig] = (() => {
   ];
 })();
 
+let federatedContentRoot;
```

Comment:
Since we are very protective of utils, here's the rationale:

`federated.js` is required for the LCP in a lot of cases due to being used in `decorate.js`

On top of this, it has wide usage across multiple blocks that are being used for LCP.

Anything that we absolutely need for the 90% LCP use-case and critical to rendering the page is justified to live in the utils in my opinion. On top of that, wide usage across blocks makes this a perfect candidate to live here.

Reply:
This does sound reasonable, but do you have any data to support that this is the better approach performance-wise?

On another note, would we need to move the unit tests in the utils test file?

Reply:
We saw a 400ms LCP improvement when we merged https://github.com/adobecom/milo/pull/2752#issue-2470375361 which was a combination of parallelize 2 requests & get rid of one. This should net us ~200ms for all devices.

For the unit tests, since we still have the original file, for now we can keep them where they are IMO 🤔 
The unit test-tests are full enough anyway so the separation is not bad